src/core/lib/transport/call_spine.h

```diff
@@ -191,6 +196,71 @@ class CallSpine final : public Party {
     });
   }
 
+  // Spawned operations: these are callable from /outside/ the call; they spawn
+  // an operation into the call and execute that operation.
+
+  void SpawnPushServerInitialMetadata(ServerMetadataHandle md) {
+    SpawnInfallible(
+        "push-server-initial-metadata",
+        [md = std::move(md), self = RefAsSubclass<CallSpine>()]() mutable {
+          self->CancelIfFailed(self->PushServerInitialMetadata(std::move(md)));
+        });
+  }
+
+  auto SpawnPushServerToClientMessage(MessageHandle msg) {
+    return SpawnWaitable(
+        "push-message",
+        [msg = std::move(msg), self = RefAsSubclass<CallSpine>()]() mutable {
+          return self->CancelIfFails(
+              self->PushServerToClientMessage(std::move(msg)));
+        });
+  }
+
+  auto SpawnPushClientToServerMessage(MessageHandle msg) {
+    return SpawnWaitable(
+        "push-message",
+        [msg = std::move(msg), self = RefAsSubclass<CallSpine>()]() mutable {
+          return self->CancelIfFails(
+              self->PushClientToServerMessage(std::move(msg)));
+        });
+  }
+
+  void SpawnFinishSends() {
+    SpawnInfallible("finish-sends", [self = RefAsSubclass<CallSpine>()]() {
+      self->FinishSends();
+      return Empty{};
+    });
+  }
+
+  void SpawnPushServerTrailingMetadata(ServerMetadataHandle md) {
+    SpawnInfallible(
+        "push-server-trailing-metadata",
+        [md = std::move(md), self = RefAsSubclass<CallSpine>()]() mutable {
+          self->PushServerTrailingMetadata(std::move(md));
+          return Empty{};
+        });
+  }
+
+  void AddChildCall(RefCountedPtr<CallSpine> child_call) {
+    child_calls_.push_back(std::move(child_call));
+    if (child_calls_.size() == 1) {
+      SpawnInfallible("check_cancellation",
+                      [self = RefAsSubclass<CallSpine>()]() mutable {
+                        auto was_cancelled = self->WasCancelled();
+                        return Map(std::move(was_cancelled),
+                                   [self = std::move(self)](bool cancelled) {
+                                     if (cancelled) {
+                                       for (auto& child : self->child_calls_) {
+                                         child->SpawnPushServerTrailingMetadata(
+                                             CancelledServerMetadataFromStatus(
+                                                 absl::CancelledError()));
```

Comment:
Is this always the right status to use here?  Are there any code-paths where cancellation can occur with a different status, which we might want to propagate?

I suspect this is a non-issue, but asking just to be sure.

Reply:
I think this is a non-issue.. we're setting the trailing metadata value for an attempt that we'll never propagate up.. this is more just to shake any infrastructure that's waiting for a message or such out of what it's doing and into the cleanup path.